### PR TITLE
Alert Dialog Example: Use icon and button description to reflect save status instead of toast alert

### PR DESCRIPTION
**Pull request by howard-e**
Addresses #756.

[Preview updated alert dialog example](https://raw.githack.com/w3c/aria-practices/bocoup/replace-dialog-alert-save-toast/examples/dialog-modal/alertdialog.html)

### CODE REVIEW

File: examples/dialog-modal/alertdialog.html [hunk 1]
```diff
@@ -68,9 +68,17 @@ <h2 id="ex_label">Example</h2>
...
-          <button type="button" aria-controls="notes" id="notes_save">Save</button>
+          <div class="visually-hidden" id="notes_save_status" role="alert"></div>
+          <button type="button" aria-controls="notes" id="notes_save">
+            Save
+            <svg class="icon spinner" viewBox="0 0 32 32">
+              <path d="M16 32c-4.274 0-8.292-1.664-11.314-4.686s-4.686-7.040-4.686-11.314c0-3.026 0.849-5.973 2.456-8.522 1.563-2.478 3.771-4.48 6.386-5.791l1.344 2.682c-2.126 1.065-3.922 2.693-5.192 4.708-1.305 2.069-1.994 4.462-1.994 6.922 0 7.168 5.832 13 13 13s13-5.832 13-13c0-2.459-0.69-4.853-1.994-6.922-1.271-2.015-3.066-3.643-5.192-4.708l1.344-2.682c2.615 1.31 4.824 3.313 6.386 5.791 1.607 2.549 2.456 5.495 2.456 8.522 0 4.274-1.664 8.292-4.686 11.314s-7.040 4.686-11.314 4.686z"></path>
+            </svg>
+            <svg class="icon check" viewBox="0 0 32 32">
+              <path d="M27 4l-15 15-7-7-5 5 12 12 20-20z"></path>
+            </svg>
```
Comment (zcorpan) [inline examples/dialog-modal/alertdialog.html:79]: Should the `svg`s have `aria-hidden="true"`?

**Comment by zcorpan** [comment]
cc @smhigley @a11ydoer

**Comment by mcking65** [comment]
@howard-e 

Per our discussion on Nov 2, please put disabled save button in tab sequence by replacing disabled attribute with aria-disabled="true".

After save, focus is getting lost; I think it is going to body instead of to the save button. This is probably because of the disabled attribute.

File: examples/dialog-modal/alertdialog.html [hunk 2]
```diff
@@ -68,9 +68,17 @@ <h2 id="ex_label">Example</h2>
         <div id="ex_alertdialog">
           <label for="notes">Notes</label>
           <textarea class="notes" name="notes" id="notes" rows="7">This is an example text box, which unsurprisingly contains text. The user is given the option to save this text - which triggers a basic alert - or to discard the text - which triggers an alert dialog that prompts the user for confirmation.</textarea>
-          <button type="button" aria-controls="notes" id="notes_save">Save</button>
+          <div class="visually-hidden" id="notes_save_status" role="alert"></div>
+          <button type="button" aria-controls="notes" id="notes_save">
```
Comment (mcking65) [inline examples/dialog-modal/alertdialog.html:72]: This should not have aria-controls

File: examples/dialog-modal/alertdialog.html [hunk 3]
```diff
@@ -68,9 +68,17 @@ <h2 id="ex_label">Example</h2>
...
+            Save
+            <svg class="icon spinner" viewBox="0 0 32 32">
+              <path d="M16 32c-4.274 0-8.292-1.664-11.314-4.686s-4.686-7.040-4.686-11.314c0-3.026 0.849-5.973 2.456-8.522 1.563-2.478 3.771-4.48 6.386-5.791l1.344 2.682c-2.126 1.065-3.922 2.693-5.192 4.708-1.305 2.069-1.994 4.462-1.994 6.922 0 7.168 5.832 13 13 13s13-5.832 13-13c0-2.459-0.69-4.853-1.994-6.922-1.271-2.015-3.066-3.643-5.192-4.708l1.344-2.682c2.615 1.31 4.824 3.313 6.386 5.791 1.607 2.549 2.456 5.495 2.456 8.522 0 4.274-1.664 8.292-4.686 11.314s-7.040 4.686-11.314 4.686z"></path>
+            </svg>
+            <svg class="icon check" viewBox="0 0 32 32">
+              <path d="M27 4l-15 15-7-7-5 5 12 12 20-20z"></path>
+            </svg>
+          </button>
           <button type="button" aria-controls="notes" id="notes_discard" onclick="openAlertDialog('alert_dialog', this)">Discard</button>
```
Comment (mcking65) [inline examples/dialog-modal/alertdialog.html:81]: This should not have aria-controls

**Review by mcking65 — CHANGES_REQUESTED** [review]

**Comment by howard-e** [comment]
> Per our discussion on Nov 2, please put disabled save button in tab sequence by replacing disabled attribute with aria-disabled="true".

Thanks! Unfortunately, I had missed that meeting so thanks for that note.

File: examples/dialog-modal/alertdialog.html (hunk 2 above)
Comment (a11ydoer) [reply examples/dialog-modal/alertdialog.html]: @smhigley will push change in correcting the case - removing aria-control broke the dialog.

**Comment by smhigley** [comment]
@howard-e I just pushed the commit we talked about in the APG meeting that removes aria-controls and fixes the JS that depended on it. Also updated it so that the `aria-disabled` button won't activate when clicked while I was in there, since I saw that was a recent change + easy to miss :).

**Comment by howard-e** [comment]
Thanks @smhigley!

**Review by mcking65 — APPROVED** [review]
Editorial review complete; approving all copy.

**Comment by mcking65** [comment]
@howard-e  or @jongund 

I looked at the conflicts in the css but have 0 confidence that I can resolve without unknowingly breaking anything. It looks like something that would take only a few minutes. Either of you available to add a commit with a fix for that conflict?

**Comment by mcking65** [comment]
It looks like we are missing a lot of tests for this example.

**Comment by howard-e** [comment]
> I looked at the conflicts in the css but have 0 confidence that I can resolve without unknowingly breaking anything. It looks like something that would take only a few minutes. Either of you available to add a commit with a fix for that conflict?

@mcking65 Done!

**Comment by mcking65** [comment]
@jongund  Do you have a moment to take a final look at this morning?

I'm planning to merge later this morning, would be good to have a second person make a final check of visuals.

**Review by jongund — APPROVED** [review]
Looks good to me.

**Comment by jongund** [comment]
@mcking65 Sent an approving review.

**Pull request merged**